python/lsst/jointcal/ccdImage.cc

```diff
@@ -66,13 +65,11 @@ void declareCcdImageList(py::module &mod) {
 
     cls.def("sizeValidForFit", &CcdImageList::sizeValidForFit);
 
-    // Make this behave like a std::list, which it is, but we can't wrap children of it easily.
+    // Make this behave like a std::list (which it is but we can't wrap std::list subclasses trivially)
```

Comment:
FWIW, I don't think we should remove custom containers just on principle.  Sometimes doing that can really add value if the container has some special functionality, and it can be a nice solution to the case where we *can't* convert directly to a built-in Python container in the wrapper layer for one reason or another.

Of course, any custom containers *should* delegate most of their work to an STL container (and do so via composition, not inheritance).  I have no idea if these particular containers add value over their STL counterparts, but I do suspect Pierre had a good reason to choose `std::list` over `std::vector`.